.paw/work/paw-review/ImplementationPlan.md

```diff
@@ -0,0 +1,493 @@
+
+## Desired End State
+A set of agent chatmodes and supporting prompts implementing FR-001..FR-034 with artifacts stored under `.paw/reviews/<identifier>/` (PR-N or branch-slug). Reviewer can invoke each stage; artifacts generated in correct dependency order; GitHub pending review (comments only) created; ReviewComments.md contains rationale and assessments; supports regeneration and tone adjustment; no open questions; all success criteria SC-001..SC-016 verifiable.
+
+### Key Discoveries:
+- Chatmode pattern & metadata frontmatter (existing agents)
+- WorkflowContext parameter extraction pattern
+- Research modes (location, analysis, pattern finder) inform CodeResearch reuse
+- Pending review creation via `pull_request_review_write` + `add_comment_to_pending_review`
+- One Issue, One Comment batching policy in specification
+
+## What We're NOT Doing
+- Automatic submission or merging of PRs
+- Batch reviews across multiple PRs simultaneously
+- Machine learning prioritization or tone personalization profiles
+- Extending beyond Must/Should/Could categorization
+- Implementing coverage instrumentation tooling (will consume existing metrics only)
+
+## Implementation Approach
+Incrementally add new chatmode agent files and shared prompt templates. Reuse existing helper patterns (frontmatter gathering, parameter parsing). Create artifact generator logic inside agents referencing spec requirements; ensure each artifact fully self-contained. Provide internal utility functions (or procedural steps) for: PR metadata fetch, diff/stat retrieval, file categorization, mechanical vs semantic classification, change graph building, baseline code research invocation, impact traversal, test coverage parsing, comment batching, tone regeneration, assessment injection, Q&A referencing artifacts.
+
+## Phase 1: Understanding Stage Agent & Artifacts
+### Overview
+Implement the Understanding stage producing: `PRContext.md`, `prompts/code-research.prompt.md`, `CodeResearch.md` (generated by separate research agent after pause), `DerivedSpec.md`, `ChangeAnalysis.md`. Establish baseline evidence before evaluation.
+
+### Components / Files
+1. Chatmode: `.github/chatmodes/PAW-R1 Understanding Agent.chatmode.md`
+2. Artifacts Directory (GitHub PR): `.paw/reviews/PR-<number>/` ; Non-GitHub: `.paw/reviews/<branch-slug>/`
+3. Artifacts:
+	- `PRContext.md`
+	- `prompts/code-research.prompt.md`
+	- `CodeResearch.md` (post-research)
+	- `DerivedSpec.md`
+	- `ChangeAnalysis.md`
+
+### Data Flow
+Input: PR URL OR (base branch, current head branch) + `WorkflowContext.md`.
```

Comment:
WorkflowContext.md will not be available. It's just the PR or branch.
We should rename `PRContext.md` to `ReviewContext.md`, and each stage will use that as the primary entry point artifact, similar to how the implementation PAW process uses WorkflowContext.md.